lnwallet/commitment.go

```diff
@@ -610,30 +612,108 @@ func CommitScriptAnchors(chanType channeldb.ChannelType,
 	return localAnchor, remoteAnchor, nil
 }
 
+// CommitSortFunc is a function type alias for a function that sorts the
+// commitment transaction outputs. The second parameter is a list of CLTV
+// timeouts that must correspond to the number of transaction outputs, with the
+// value of 0 for non-HTLC outputs.
+type CommitSortFunc func(*wire.MsgTx, []uint32, []input.HtlcIndex) error
```

Comment:
Do you mean not using mutation to do the sort? FWIW that's how things are with the official sort function: https://pkg.go.dev/sort#Sort

Reply:
I'm talking about sorting the htlcs according to the sort rules before rendering them to outputs, rather than rendering to outputs and using parallel vectors to do an in-place sort.

Reply:
Maybe I'll just clean this up myself at a later date though.

Reply:
Yeah, I think this would explode the scope of this PR. Though I agree this could probably be done in a better way.